libs/core/errors/src/exception.cpp

```diff
@@ -329,8 +333,7 @@ namespace hpx { namespace detail {
         hpx::exception const&, std::string const&, std::string const&, long);
 
     template HPX_CORE_EXPORT void throw_exception(
-        boost::system::system_error const&, std::string const&,
-        std::string const&, long);
+        std::system_error const&, std::string const&, std::string const&, long);
```

Comment:
I think you'll need to add an explicit template instantiation of `get_exception` above for `std::system_error` as well (or replace the one with `boost::system::system_error`?).